app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -55,7 +55,7 @@ public function hookAction()
 
             $bodyParams = json_decode(file_get_contents('php://input'), true);
 
-            if ($bodyParams['type'] == "discounts.code.apply") {
+            if (isset($bodyParams['type']) && $bodyParams['type'] == "discounts.code.apply") {
```

Comment:
can you send me oneline PR for this change?

Reply:
Done.